openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java

```diff
@@ -913,52 +913,44 @@ private MotechEvent createEventForMember(String cohortQueryUuid, CohortQueryRepo
 
     private List<Observation> createObservationGroups() {
         Concept concept1 = new Concept();
-        concept1.setUuid("child1Child1Uuid");
-
-        Observation childChild1 = new Observation();
-        childChild1.setConcept(concept1);
-        childChild1.setValue(new Observation.ObservationValue("No"));
-        childChild1.setObsDatetime(DATE);
-
-        Concept concept2 = new Concept();
-        concept2.setUuid("child2Child1Uuid");
-
-        Observation childChild2 = new Observation();
-        childChild2.setConcept(concept2);
-        childChild2.setValue(new Observation.ObservationValue("Yes"));
-        childChild2.setObsDatetime(DATE);
-
-        Concept concept3 = new Concept();
-        concept3.setUuid("child1Uuid");
+        concept1.setUuid("child2Child1Uuid");
 
         Observation child1 = new Observation();
-        child1.setConcept(concept3);
-        child1.setGroupMembers(Arrays.asList(childChild2, childChild1));
+        child1.setConcept(concept1);
+        child1.setValue(new Observation.ObservationValue("Yes"));
         child1.setObsDatetime(DATE);
 
-        Concept concept4 = new Concept();
-        concept4.setUuid("child2Uuid");
+        Concept concept2 = new Concept();
+        concept2.setUuid("child1Child1Uuid");
 
         Observation child2 = new Observation();
-        child2.setConcept(concept4);
-        child2.setValue(new Observation.ObservationValue("2nd"));
+        child2.setConcept(concept2);
+        child2.setValue(new Observation.ObservationValue("No"));
         child2.setObsDatetime(DATE);
 
-        Concept concept5 = new Concept();
-        concept5.setUuid("child3Uuid");
+        Concept concept3 = new Concept();
+        concept3.setUuid("child2Uuid");
 
         Observation child3 = new Observation();
-        child3.setConcept(concept5);
-        child3.setValue(new Observation.ObservationValue("Maybe"));
+        child3.setConcept(concept3);
+        child3.setValue(new Observation.ObservationValue("2nd"));
         child3.setObsDatetime(DATE);
 
-        Concept concept6 = new Concept();
-        concept6.setUuid("parentUuid");
+        Concept concept4 = new Concept();
+        concept4.setUuid("child3Uuid");
+
+        Observation child4 = new Observation();
+        child4.setConcept(concept4);
+        child4.setValue(new Observation.ObservationValue("Maybe"));
+        child4.setObsDatetime(DATE);
+
+        Concept concept5 = new Concept();
+        concept5.setUuid("parentUuid");
 
         Observation parentObservation = new Observation();
         parentObservation.setObsDatetime(DATE);
-        parentObservation.setConcept(concept6);
-        parentObservation.setGroupMembers(Arrays.asList(child1, child2, child3));
+        parentObservation.setConcept(concept5);
+        parentObservation.setGroupMembers(Arrays.asList(child1, child2, child3, child4));
```

Comment:
@kmadej OpenMRS, for some reason, is ignoring nested groupMembers. When I'm creating only the observations which are containing some value (last UUID in group), they're created.